plc4j/drivers/opcua/src/main/java/org/apache/plc4x/java/opcua/connection/OpcuaTcpPlcConnection.java

```diff
@@ -0,0 +1,469 @@
+            return new DefaultShortFieldItem((Short)objValue);
+        }else if (typeNode.equals(Identifiers.Int32)){
+            return new DefaultIntegerFieldItem((Integer)objValue);
+        }else if (typeNode.equals(Identifiers.Int64)){
+            return new DefaultLongFieldItem((Long)objValue);
+        }else if (typeNode.equals(Identifiers.UInteger)){
+            return new DefaultLongFieldItem((Long)objValue);
+        }else if (typeNode.equals(Identifiers.UInt16)){
+            return new DefaultIntegerFieldItem(((UShort)objValue).intValue());
+        }else if (typeNode.equals(Identifiers.UInt32)){
+            return new DefaultLongFieldItem(((UInteger)objValue).longValue());
+        }else if (typeNode.equals(Identifiers.UInt64)){
+            return new DefaultBigIntegerFieldItem(new BigInteger(objValue.toString()));
+        }else if (typeNode.equals(Identifiers.Byte)){
+            return new DefaultShortFieldItem(Short.valueOf(objValue.toString()));
+        }else if (typeNode.equals(Identifiers.Float)){
+            return new DefaultFloatFieldItem((Float)objValue);
+        }else if (typeNode.equals(Identifiers.Double)){
+            return new DefaultDoubleFieldItem((Double)objValue);
+        }else if (typeNode.equals(Identifiers.SByte)){
+            return new DefaultByteFieldItem((Byte)objValue);
+        }else {
+            return new DefaultStringFieldItem(objValue.toString());
+        }
+
+    }
+
+    @Override
+    public CompletableFuture<PlcWriteResponse> write(PlcWriteRequest writeRequest) {
+        CompletableFuture<PlcWriteResponse> future;
+        future = CompletableFuture.supplyAsync(() -> {
+
+            InternalPlcWriteRequest internalPlcWriteRequest = (InternalPlcWriteRequest) writeRequest;
+
+            List<PlcField> writePLCValues = writeRequest.getFields();
+            LinkedList<DataValue> values = new LinkedList<>();
+            LinkedList<NodeId> ids = new LinkedList<>();
+            LinkedList<String> names = new LinkedList<>();
+            Map<String, PlcResponseCode> fieldResponse = new HashMap<>();
+            for (String fieldName: writeRequest.getFieldNames()) {
+                OpcuaField uaField = (OpcuaField) writeRequest.getField(fieldName);
+                NodeId idNode = generateNodeId(uaField);
+                Variant var = new Variant(internalPlcWriteRequest.getFieldItem(fieldName).getObject(0));
+                DataValue value = new DataValue(var, null, null);
+                ids.add(idNode);
+                names.add(fieldName);
+                values.add(value);
+            }
+            CompletableFuture<List<StatusCode>> opcRequest =
+                client.writeValues(ids, values);
+            List<StatusCode> statusCodes = null;
+            try {
+                statusCodes = opcRequest.get();
+            } catch (InterruptedException | ExecutionException e) {
+                statusCodes = new LinkedList<>();
```

Comment:
We should also forward this somehow to the user

Reply:
I will first add everywhere some warnings where we can thing afterwards how PLC4J should behave in generall? or is there some doc about the general behavior of the API?